flow/Makefile

```diff
@@ -158,9 +166,37 @@ ifeq (, $(strip $(NPROC)))
 endif
 export NUM_CORES := $(NPROC)
 
-OPENROAD_CMD = openroad -no_init -exit
-OPENROAD_NO_EXIT_CMD = $(patsubst -exit,,$(OPENROAD_CMD))
+#-------------------------------------------------------------------------------
+# setup all commands used within this flow
+TIME_CMD = /usr/bin/time -f "%Eelapsed %PCPU %MmemKB"
+TIME_TEST = $(shell $(TIME_CMD) echo foo 2>/dev/null)
+ifeq (, $(strip $(TIME_TEST)))
+  TIME_CMD = /usr/bin/time
+endif
 
+# The following determine the executable location for each tool used by this flow.
+# Priority is given to 
+#       1 user include path to the executable prior to running make
+#       2 user explicity set with variable in Makefile or command line
+OPENROAD_CMD            ?= $(shell command -v openroad)
+ifeq ($(OPENROAD_CMD),)
+  OPENROAD_CMD           = $(abspath $(OPENROAD_HOME)/../tools/build/OpenROAD/src/openroad) -no_init -exit
+else
+  OPENROAD_CMD           = $(shell command -v openroad) -no_init -exit
+endif
+OPENROAD_NO_EXIT_CMD     = $(firstword $(OPENROAD_CMD)) -no_exit
+
+YOSYS_CMD               ?= $(shell command -v yosys)
+ifeq ($(YOSYS_CMD),)
+  YOSYS_CMD              = $(abspath $(OPENROAD_HOME)/../tools/build/yosys/bin/yosys)
+endif
+
+KLAYOUT_CMD             ?= $(shell command -v klayout)
+ifeq ($(KLAYOUT_CMD),)
+  $(error KLayout command not found.Please configure KLayout prior to running OpenROAD's flow)
```

Comment:
The flow can still run, just won't be able to generate the gds.
I would suggest that is a warning, since it's not a dealbreaker if all you want is the def file.